### PR TITLE
Fix url insert on the top menu

### PR DESCRIPTION
**Pull request by AntoineMille**
Please see : http://forge.prestashop.com/browse/BOOM-290

### CODE REVIEW

File: ps_menutoplinks.class.php [hunk 1]
```diff
@@ -67,6 +67,16 @@ public static function getLinkLang($id_linksmenutop, $id_shop)
         return array('link' => $link, 'label' => $label, 'new_window' => $new_window);
     }
 
+    public static function updateUrl($link)
+    {
+        for($i = 1; $i <= count($link); $i++) {
+            if (substr($link[$i], 0, 4) === "www.") {
```
Comment (maximebiloe) [inline ps_menutoplinks.class.php]: What if my url does not contain `www`?
I think it should be better to test if `http://` is missing, no?

**Comment by AntoineMille** [comment]
Hi @maximebiloe,

For me It's good now.

Thanks

File: ps_menutoplinks.class.php [hunk 2]
```diff
@@ -67,6 +67,16 @@ public static function getLinkLang($id_linksmenutop, $id_shop)
         return array('link' => $link, 'label' => $label, 'new_window' => $new_window);
     }
 
+    public static function updateUrl($link)
+    {
+        for($i = 1; $i <= count($link); $i++) {
+            if (substr($link[$i], 0, 7) !== "http://" && substr($link[$i], 0, 8) !== "https://") {
+                $link[$i] = "http://" . $link[$i];
```
Comment (aleeks) [inline ps_menutoplinks.class.php]: You had http, but if the server is using https, it could be not good. See if you can find some var on $_SERVER ? WDYT ?

Comment (AntoineMille) [inline ps_menutoplinks.class.php]: Hi @aleeks, I don't understand the usefulnesss of using the variable $_SERVER in this context.
I think it's a good idea to use directly "http://" and "https://" for this.

Comment (aleeks) [inline ps_menutoplinks.class.php]: Ok, i don't understand you want use external url. Keep in mind, sometime, it will not work.
Just add a text for the user!

Comment (AntoineMille) [inline ps_menutoplinks.class.php]: So, I added a placeholder (Thanks @AlexEven) but I kept the verification.

**Comment by aleeks** [comment]
@AntoineMille need rebase hihi

**Comment by AntoineMille** [comment]
@aleeks It's GOOD now :dagger:

**Comment by maximebiloe** [comment]
Thank you @AntoineMille

**Comment by Askadar** [comment]
Hi there, I've noticed a problem with your fix - it breaks relative urls. 

TLDR: 
change updateUrl() condition to `substr($link[$i], 0, 7) !== "http://" && substr($link[$i], 0, 8) !== "https://" && substr($link[$i], 0, 1) !== '/'`
i.e. if link doesn't have http(s?) AND it's not a relative to site's root, "fix" it.

For example I have wesite `http://example.com`, on which I've enabled seo urls and got myself a product with category that got url like `http://example.com/category/2-product`. All nice and good, but now I noticed that it's very popular, so I want to put it on my slider (ps_imageslider module) and even inside top menu (ps_mainmenu), being smartypants guy I decide to futerproof my possible switch to https and maybe even domain name (cause, I'll be honest, example.com isn't going to cut it in the long run) so I've used relative URI - `/category/2-product` but, when I added it to my slider, this fix broke it (since it doesn't have `http(s?)://` in the beggining) into some monstrosity - `http:///category/product` which isn't even a valid URI.

So, right now I propose simple logic addition: 
`substr($link[$i], 0, 7) !== "http://" && substr($link[$i], 0, 8) !== "https://" && substr($link[$i], 0, 1) !== '/'`
i.e. if link doesn't have http(s?) AND it's not relative to site's root, "fix" it.

It'll work for now and be nice and dandy, but, we will probably need a proper hostname resolver, yada-yada-yada.
I can create separate issues on megamenu and imageslider modules, those are two I've found to have this problem, in case that's needed.
